airbyte-integrations/connectors/source-db2/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/Db2SourceDatatypeTest.java

```diff
@@ -126,7 +127,7 @@ protected void initTests() {
             .airbyteType(JsonSchemaType.NUMBER)
             .fullSourceDataType("DECIMAL(31, 0)")
             .addInsertValues("null", "1", "DECIMAL((-1 + 10E+29), 31, 0)", "DECIMAL((1 - 10E+29), 31, 0)")
-            .addExpectedValues(null, "1", "1.0E30", "-1.0E30")
+            .addExpectedValues(null, "1", "%.0f".formatted(Double.valueOf("1.0E30")), "%.0f".formatted(Double.valueOf("-1.0E30")))
```

Comment:
If I'm understanding this correctly, this will be showing integer values _only_, right? I haven't seen the usage of `%.0f` before, normally it would have one decimal precision value.

Also this seems to work but curious if this should be 
```java
Float.valueOf("1.0E30")
```
since the string formatting is indicating a Float `f` vs Double `d`

Reply:
The weeds of floating point string format. 
How I missed thee… :-)

`f` signifies a floating point number - a `d` modifier is a decimal **integer** I think, but it won't accept a double or float value.

the `%.0f` formats a floating point number to print with 0 decimal digits (0 precision) - which is the expected string.

And a `Float.valueOf("1.0E30")` is theoretically ok, but depending on actual hardware and implementation won't print as a clean `1` with zeroes but some weird floating point "1000000015047466200000000000000" and so on- it's the whole single/double precision implementation of respecting float/double types.